cmd/vulcan-aws-subdomain-takeover/cloudinventory.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2025 Adevinta
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"text/template"
+)
+
+type CloudInventory struct {
+	client   *http.Client
+	token    string
+	endpoint string
+}
+
+func NewCloudInventory(token string, endpoint string) *CloudInventory {
+	return &CloudInventory{
+		client:   &http.Client{},
+		token:    token,
+		endpoint: endpoint,
+	}
+}
+
+func (ci *CloudInventory) IsIPPublicInInventory(ip string) (bool, error) {
+	t, err := template.New("isInInventory").Parse(ci.endpoint)
+	if err != nil {
+		return false, err
+	}
```

Comment:
Move this logic to `NewCloudInventory` and store the parsed template in `CloudInventory`